wrapper/src/main/java/software/amazon/jdbc/plugin/AbstractConnectionPlugin.java

```
@@ -75,8 +75,9 @@ public boolean acceptsStrategy(HostRole role, String strategy) {
   }
 
   @Override
-  public HostSpec getHostSpecByStrategy(final HostRole role, final String strategy) {
-    return null;
+  public HostSpec getHostSpecByStrategy(final HostRole role, final String strategy)
+      throws UnsupportedOperationException {
+    throw new UnsupportedOperationException("getHostSpecByStrategy is not supported by this plugin.");
```

We shouldn't get here because plugins should be calling acceptsStrategy before getHostSpecByStrategy to assess if a strategy is valid, but if we do get here I think its better to fail fast than quietly return  null